pkg/services/publicdashboards/database/database.go

```diff
@@ -106,74 +104,39 @@ func (d *PublicDashboardStoreImpl) GetPublicDashboardAndDashboard(ctx context.Co
 	return pdRes, dashRes, err
 }
 
-// Generates a new unique uid to retrieve a public dashboard
-func (d *PublicDashboardStoreImpl) GenerateNewPublicDashboardUid(ctx context.Context) (string, error) {
-	var uid string
-
-	err := d.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
-		for i := 0; i < 3; i++ {
-			uid = util.GenerateShortUID()
-
-			exists, err := sess.Get(&PublicDashboard{Uid: uid})
-			if err != nil {
-				return err
-			}
-
-			if !exists {
-				return nil
-			}
-		}
+// GetPublicDashboardByUid Returns public dashboard configuration by Uid or nil if not found
+func (d *PublicDashboardStoreImpl) GetPublicDashboardByUid(ctx context.Context, uid string) (*PublicDashboard, error) {
+	if uid == "" {
+		return nil, nil
```

Comment:
We've been returning missing identifier error here previously. Should we do that or change them all to return nil?

Reply:
I think this is confusing because of the refactor but this function was returning `nil, nil` before, however, we should be consistent with `GetPublicDashboardByAccessToken` so I will add this return here too. Good catch!

Reply:
I had to revert this change because one of the callers is actually expecting to receive a `*PublicDashboard` nil when calling with empty `uid`. We could change it but I would do it in another PR.

Reply:
Sounds good to me!